cloudrail/knowledge/utils/port_utils.py

```
@@ -413,3 +414,17 @@ def get_overlapping_ports(first_range: Tuple[int, int], second_range: Tuple[int,
         return None
     else:
         return max_start, min_end
+
+
+def convert_ip_protocol_to_protocol_name(ip_protocol_num: Union[str, int]) -> Optional[str]:
+    if ip_protocol_num in ('-1', -1):
+        return 'all'
+    elif not ip_protocol_num.isdigit():
+        return ip_protocol_num
+    else:
+        ip_protocol_table = {num: name.replace('IPPROTO_', '') for name, num in vars(socket).items() if name.startswith("IPPROTO")}
+        try:
+            return ip_protocol_table[int(ip_protocol_num)].lower()
+        except Exception as ex:
+            print(f'fail converting ip_protocol number for {ip_protocol_num} : {ex}')
```

use logger instead of print

---

agree , u should never do print()
i dont understand what u do incase of exception. for now, it seems your return none. u need to either raise the exception and crash the process or return value the system can work with (For example all or 0 or -1)

---

@Tomerva @Viperain , I have changed it to the following:
        except Exception as ex:
            logging.exception(f'fail converting ip_protocol number for {ip_protocol_num} : {ex}')
        return str(ip_protocol_num)